homeassistant/components/zone/__init__.py

```diff
@@ -104,29 +107,57 @@ def in_zone(zone, latitude, longitude, radius=0):
     return zone_dist - radius < zone.attributes[ATTR_RADIUS]
 
 
-@asyncio.coroutine
-def async_setup(hass, config):
-    """Set up the zone."""
-    entities = set()
-    tasks = []
+async def async_setup(hass, config):
+    """Import new configured zone as config entry."""
+    zones = set()
+    zone_entries = configured_zones(hass)
     for _, entry in config_per_platform(config, DOMAIN):
-        name = entry.get(CONF_NAME)
-        zone = Zone(hass, name, entry[CONF_LATITUDE], entry[CONF_LONGITUDE],
-                    entry.get(CONF_RADIUS), entry.get(CONF_ICON),
-                    entry.get(CONF_PASSIVE))
+        name = slugify(entry[CONF_NAME])
+        if name not in zone_entries:
+            zones.add(name)
+            hass.async_add_job(hass.config_entries.flow.async_init(
+                DOMAIN, source='import', data=entry
+            ))
+
+    if HOME_ZONE not in zones and HOME_ZONE not in zone_entries:
+        entry = {
+            CONF_NAME: hass.config.location_name,
+            CONF_LATITUDE: hass.config.latitude,
+            CONF_LONGITUDE: hass.config.longitude,
+            CONF_RADIUS: DEFAULT_RADIUS,
+            CONF_ICON: ICON_HOME,
+            CONF_PASSIVE: False,
+            CONF_ENTITY_ID: ENTITY_ID_HOME
+        }
+        config_entry = ConfigEntry(1, DOMAIN, 'Home', entry, 'home-assistant')
+        hass.async_add_job(async_setup_entry(hass, config_entry))
```

Comment:
@balloob is this too hackish? Maybe you have a better suggestion on how not to cache the hass home zone?

Reply:
Too hackish. The solution is to allow setting up an entity without the config entry.

Now that I think of it, we shouldn't even import the configuration.yaml values into config entries. Just set it up.

Reply:
OK, I am not completely following you. How would you like it instead?

Reply:
This is the only part I feel is left

Reply:
The way you set up the Home zone, that is how we should also setup all configured entries in `configuration.yaml`. We should not import them as config entries.

Reply:
Ok, so I should just revert that part?

Reply:
Yes

Reply:
Followed your directions, also improved some tests. Hopefully done 👍